hazelcast/src/main/java/com/hazelcast/config/BitmapIndexOptions.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import com.hazelcast.internal.config.ConfigDataSerializerHook;
+import com.hazelcast.internal.util.StringUtil;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.query.QueryConstants;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+import static com.hazelcast.internal.util.Preconditions.checkNotNull;
+
+/**
+ * Configures indexing options specific to bitmap indexes.
+ */
+public class BitmapIndexOptions implements IdentifiedDataSerializable {
+
+    /**
+     * The default for {@link #getUniqueKey() unique key}.
+     */
+    public static final String DEFAULT_UNIQUE_KEY = QueryConstants.KEY_ATTRIBUTE_NAME.value();
+
+    /**
+     * The default for {@link #getUniqueKeyTransformation() unique key
+     * transformation}.
+     */
+    public static final UniqueKeyTransformation DEFAULT_UNIQUE_KEY_TRANSFORMATION = UniqueKeyTransformation.OBJECT;
+
+    /**
+     * Defines an assortment of transformations which can be applied to {@link
+     * BitmapIndexOptions#getUniqueKey() unique key} values.
+     */
+    public enum UniqueKeyTransformation {
```

Comment:
After some thinking, I am still concerned with the enum and it's naming. We do not really do any transformation. Instead, we _map_ the key to some other value, based on its _range of values_. There are three choices: non-integer type, integer type, integer type with non-negative values. Current naming - `OBJECT`, `LONG` and `RAW` respectively - doesn't reflect that. Another problem is that these names are Java-centric (e.g. they assume the `Object` type as a root of a type hierarchy), and we are trying to get rid of that on the product level.

I would think about different naming, e.g.:
```
enum KeyValueRange {
    ANY,
    WHOLE_NUMBER,
    NATURAL_NUMBER
}
```

Reply:
After the internal discussion, we have decided to keep the names as they are now, because neither of the naming schemes provide significant advantages.